### ISSUE TITLE
Add more context to `not_a_type` diagnostics

### ISSUE THREAD

**Issue by natebosch** [dart-lang/sdk#55947]
Some time back the analyzer improved `argument_type_not_assignable` (and I'd imagine others) diagnostics to include more context about the types. Specifically it will say something like  "where WhateverType is defined in `URI`".

Can we add similar context to `not_a_type` diagnostics? There should be a reference to a non-Type that we can point to.

**Comment by srawlins** [comment]
CC @bwilkerson

**Comment by bwilkerson** [comment]
Yes, I think it would be helpful to have a context message that tells the user where the name is defined.
